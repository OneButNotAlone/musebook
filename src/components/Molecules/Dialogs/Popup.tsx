import * as React from 'react';
import styled from 'styled-components';
import { Button } from '../../Atoms/Button';

const PopupBkg = styled.div`
    position: fixed;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: rgba(0,0,0, 0.5);
    text-align: center;
`;

const PopupInner = styled.div`
    min-width: 25vw;
    max-width: 30vw;
    height: auto;
    border: 1px solid #979797;
    border-radius: 4px;
    background-color: #FFFFFF;
    box-shadow: 0 2px 4px 0 rgba(0,0,0,0.5);
    display: flex;
    flex-direction: column;
    margin: auto;
    margin-top: calc(50vh - 120px);
`;

const PopupHeader = styled.div`
    height: 54px;
    color: #666666;
    font-family: 'Bogle', sans-serif;
    font-size: 20px;
    line-height: 27px;
    font-weight: bold;
    margin: 24px 24px 16px 24px;
`;

const PopupSubtext = styled.p`
    height: 50px;
    color: #666666;
    font-family: 'Bogle', sans-serif;
    font-size: 18px;
    line-height: 25px;
    font-weight: default;
    margin: 0px 24px 24px 24px;
`;

interface PopupProps {
    msg: string;
    close: () => void;
    show: boolean;
}

export const Popup: React.SFC<PopupProps> = ({ msg, close, show }) => (

    show ?
        <PopupBkg>
            <PopupInner>
                    <PopupHeader>{msg}</PopupHeader>
                    <PopupSubtext>{'Please try again or contact app team for assistance.'}</PopupSubtext>
                    <Button style={{alignSelf: 'flex-end', marginTop: '0px', marginRight: '24px', marginBottom: '24px' }} onClick={close}>OK</Button>
            </PopupInner>
        </PopupBkg>
        :
        null

);

export default Popup;
