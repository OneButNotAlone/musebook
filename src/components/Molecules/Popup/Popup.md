# Popup

Popups are used to show additional information or capture additional data/decisions from the user. The popup suite includes `Popup` and `Popup.Toggle`.

## Props

Prop | Type | Required | Default value | Description
--- | --- | --- | --- | ---
disableBackgroundClose | boolean | No | false | Used to disable the ability to close the popup by clicking in the background

## Usage

All children passed in `<Popup>` will make up for the content to be shown within the popup. Any component (button for example) passed within `<Popup.Toggle>` open/close the popup when clicked.

### Examples

The following code will create a button to open a popup which can be closed by clicking anywhere in the background or hitting the "Esc" key.

```html
<Popup>
    <h1>Popup content</h1>
    <p>Click anywhere in the background to close this popup</p>
</Popup>

<Popup.Toggle>
    <button>Open popup</button>
</Popup.Toggle>
```

To add a close button inside the popup, add the `Popup.Toggle` component inside the Popup as shown below.

```html
<Popup>
    <h1>Popup content</h1>
    <p>Click anywhere in the background to close this popup</p>
    <Popup.Toggle>
        <button>Close popup</button>
    </Popup.Toggle>
</Popup>

<Popup.Toggle>
    <button>Open popup</button>
</Popup.Toggle>
```

You can also disable the closing of the popup from the background or the Esc key by passing it the `disableBackgroundClose` prop with value `true` as shown below.

```html
<Popup disableBackgroundClose={true}>
    <h1>Popup content</h1>
    <p>Click anywhere in the background to close this popup</p>
    <Popup.Toggle>
        <button>Close popup</button>
    </Popup.Toggle>
</Popup>

<Popup.Toggle>
    <button>Open popup</button>
</Popup.Toggle>
```
