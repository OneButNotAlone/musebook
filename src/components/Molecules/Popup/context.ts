export const namespace: string = "aether_popup";