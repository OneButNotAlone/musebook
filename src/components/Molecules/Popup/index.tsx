import PopupContainer from './Popup.Container';
import PopupToggle from './Popup.Toggle';

export * from './Popup.Container';
export * from './Popup.Toggle';

export default Object.assign(PopupContainer, {
    Toggle: PopupToggle
});
