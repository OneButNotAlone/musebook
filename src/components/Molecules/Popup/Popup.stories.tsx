import * as React from 'react';
import { storiesOf } from '@storybook/react';

import Popup from './';

import * as PopupMarkdown from './Popup.md'

storiesOf('Popup', module)
    .add('Basic', () => (
        <>
            <Popup>
                <h1>Popup content</h1>
                <p>Click anywhere in the background to close this popup</p>
            </Popup>

            <Popup.Toggle>
                <button>Open popup</button>
            </Popup.Toggle>
        </>
    ), { notes: PopupMarkdown })
    .add('With close button', () => (
        <>
            <Popup>
                <h1>Popup content</h1>
                <p>Click anywhere in the background or the button below to close this popup</p>
                <Popup.Toggle>
                    <button>Close popup</button>
                </Popup.Toggle>
            </Popup>

            <Popup.Toggle>
                <button>Open popup</button>
            </Popup.Toggle>
        </>
    ), { notes: PopupMarkdown })
    .add('With background clicking disabled', () => (
        <>
            <Popup disableBackgroundClose={true}>
                <h1>Popup content</h1>
                <p>Closing this popup by clicking in the background is disabled</p>
                <Popup.Toggle>
                    <button>Close popup</button>
                </Popup.Toggle>
            </Popup>

            <Popup.Toggle>
                <button>Open popup</button>
            </Popup.Toggle>
        </>
    ), { notes: PopupMarkdown })
