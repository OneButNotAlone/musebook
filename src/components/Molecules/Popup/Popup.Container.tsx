import * as React from 'react';
import { Block, Overlay, Backdrop } from 'reakit';
import { namespace } from './context';

export interface PopupContainerProps {
    disableBackgroundClose?: boolean;
}

const PopupContainer: React.SFC<PopupContainerProps> = ({ disableBackgroundClose = false, children }) => {
    const allowExternalInteraction = disableBackgroundClose ? false : true;
    return (
        <Overlay.Container context={namespace}>
            {(overlay) => (
                <Block>
                    <Backdrop fade {...overlay} />
                    <Overlay
                        fade slide expand
                        {...overlay}
                        hideOnClickOutside={allowExternalInteraction}
                        hideOnEsc={allowExternalInteraction}>
                        {children}
                    </Overlay>
                </Block>
            )}
        </Overlay.Container>
    );
};

export default PopupContainer;
