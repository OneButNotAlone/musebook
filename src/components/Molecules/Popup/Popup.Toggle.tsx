import * as React from 'react';
import { Overlay, Block } from 'reakit';
import { namespace } from './context';

const PopupToggleComponent: React.SFC<{}> = (props) => {
    return (
        <Overlay.Container context={namespace}>
            {(overlay) => (
                <Block as={Overlay.Toggle} {...overlay}>
                    {props.children}
                </Block>
            )}
        </Overlay.Container>
    );
};

export default PopupToggleComponent;
