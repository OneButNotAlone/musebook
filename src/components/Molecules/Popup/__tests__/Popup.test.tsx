import * as React from 'react';
import 'jest-styled-components';
import Popup from '../';
import * as renderer from 'react-test-renderer'


test('Checked popup should render correctly', () => {
    const tree = renderer.create(
        <Popup />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Checked popup with disableBackgroundClose prop should render correctly', () => {
    const tree = renderer.create(
        <Popup disableBackgroundClose={true} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Checked Popup.Toggle should render correctly', () => {
    const tree = renderer.create(
        <Popup.Toggle />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
