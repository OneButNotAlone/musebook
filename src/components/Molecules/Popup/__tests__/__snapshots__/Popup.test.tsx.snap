// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checked Popup.Toggle should render correctly 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  display: block;
}

<button
  className="c0 c1 "
  onClick={[Function]}
/>
`;

exports[`Checked popup should render correctly 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  -webkit-transform-origin: calc(50% + 0px) calc(50% + 0px);
  -ms-transform-origin: calc(50% + 0px) calc(50% + 0px);
  transform-origin: calc(50% + 0px) calc(50% + 0px);
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
}

.c3[aria-hidden="true"] {
  pointer-events: none;
  opacity: 0;
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  visibility: hidden;
  will-change: transform,opacity;
}

.c5 {
  -webkit-transform: translate3d(-50%,-50%,0px);
  -ms-transform: translate3d(-50%,-50%,0px);
  transform: translate3d(-50%,-50%,0px);
  -webkit-transform-origin: calc(50% + 0px) calc(50% + 0px);
  -ms-transform-origin: calc(50% + 0px) calc(50% + 0px);
  transform-origin: calc(50% + 0px) calc(50% + 0px);
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
}

.c5[aria-hidden="true"] {
  pointer-events: none;
  opacity: 0;
  -webkit-transform: translate3d(-50%,calc(100% + -50%),0px) scale(0.01);
  -ms-transform: translate3d(-50%,calc(100% + -50%),0px) scale(0.01);
  transform: translate3d(-50%,calc(100% + -50%),0px) scale(0.01);
  visibility: hidden;
  will-change: transform,opacity;
}

.c2 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 998;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  -moz-tap-highlight-color: rgba(0,0,0,0);
}

.c0 {
  display: block;
}

.c4 {
  position: fixed;
  z-index: 19900410;
  left: 50%;
  top: 50%;
}

<div
  className="c0 c1"
>
  <div
    aria-hidden={true}
    className="c2 c3 c1"
    hidden={false}
    onTransitionEnd={[Function]}
    role="button"
    tabIndex={-1}
  />
  <div
    aria-hidden={true}
    aria-modal={true}
    className="c4 c5 c1"
    hidden={false}
    onTransitionEnd={[Function]}
    role="dialog"
  />
</div>
`;

exports[`Checked popup with disableBackgroundClose prop should render correctly 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  -webkit-transform-origin: calc(50% + 0px) calc(50% + 0px);
  -ms-transform-origin: calc(50% + 0px) calc(50% + 0px);
  transform-origin: calc(50% + 0px) calc(50% + 0px);
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
}

.c3[aria-hidden="true"] {
  pointer-events: none;
  opacity: 0;
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  visibility: hidden;
  will-change: transform,opacity;
}

.c5 {
  -webkit-transform: translate3d(-50%,-50%,0px);
  -ms-transform: translate3d(-50%,-50%,0px);
  transform: translate3d(-50%,-50%,0px);
  -webkit-transform-origin: calc(50% + 0px) calc(50% + 0px);
  -ms-transform-origin: calc(50% + 0px) calc(50% + 0px);
  transform-origin: calc(50% + 0px) calc(50% + 0px);
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
}

.c5[aria-hidden="true"] {
  pointer-events: none;
  opacity: 0;
  -webkit-transform: translate3d(-50%,calc(100% + -50%),0px) scale(0.01);
  -ms-transform: translate3d(-50%,calc(100% + -50%),0px) scale(0.01);
  transform: translate3d(-50%,calc(100% + -50%),0px) scale(0.01);
  visibility: hidden;
  will-change: transform,opacity;
}

.c2 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 998;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  -moz-tap-highlight-color: rgba(0,0,0,0);
}

.c0 {
  display: block;
}

.c4 {
  position: fixed;
  z-index: 19900410;
  left: 50%;
  top: 50%;
}

<div
  className="c0 c1"
>
  <div
    aria-hidden={true}
    className="c2 c3 c1"
    hidden={false}
    onTransitionEnd={[Function]}
    role="button"
    tabIndex={-1}
  />
  <div
    aria-hidden={true}
    aria-modal={true}
    className="c4 c5 c1"
    hidden={false}
    onTransitionEnd={[Function]}
    role="dialog"
  />
</div>
`;
